itou/www/apply/views/edit_views.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404, render
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from itou.job_applications.models import JobApplication
+from itou.www.apply.forms import EditHiringDateForm
+
+
+@login_required
+def edit_contract_start_date(request, job_application_id, template_name="apply/edit_contract_start_date.html"):
+    """
+    Update contract start date:
+    - at a future point in time
+    - if job_application has been accepted
+
+    If there is an approval linked to this job application, change its start date if possible
+
+    This view is kept apart from process or submit views
+    """
+    queryset = JobApplication.objects.siae_member_required(request.user)
+    job_application = get_object_or_404(queryset, id=job_application_id)
+
+    if not job_application.state.is_accepted:
+        PermissionDenied()
+
+    form = EditHiringDateForm(instance=job_application, data=request.POST or None)
+    url = reverse("apply:details_for_siae", kwargs={"job_application_id": job_application.id})
+
+    if request.method == "POST" and form.is_valid():
+        form.save()
+
+        messages.success(request, _("La période du contrat de travail a été mise à jour."))
+
+        if job_application.approval:
+            if job_application.approval.update_start_date(job_application.hiring_start_at):
```

Comment:
Là, je mettrais un `and` plutôt que un `if` dans un `if`.

Reply:
Carrément, oui